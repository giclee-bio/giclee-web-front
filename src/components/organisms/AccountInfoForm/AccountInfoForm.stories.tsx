import { Meta, StoryObj } from '@storybook/react'

import AccountInfoForm from './AccountInfoForm'

const meta: Meta<typeof AccountInfoForm> = {}

export default meta
type Story = StoryObj<typeof AccountInfoForm>

export const Index: Story = {
  args: {},
}
