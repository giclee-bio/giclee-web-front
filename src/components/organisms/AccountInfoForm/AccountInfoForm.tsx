import Form from '@/components/molecules/Form'

import type { Props as ButtonProps } from '@/components/atoms/Button'
import type { FormField } from '@/components/molecules/Form'

const buttonProps: ButtonProps = {
  children: '更新する',
  type: 'primary',
  width: 'half',
}

const formFields: FormField[] = [
  {
    label: '表示名',
    name: 'displayName',
    require: true,
    tag: 'input',
    type: 'text',
  },
  {
    cols: 30,
    label: '自己紹介',
    name: 'username',
    require: true,
    rows: 5,
    tag: 'textarea',
  },
]

const AccountInfoForm = () => {
  return <Form buttonCenter={false} buttonProps={buttonProps} formFields={formFields} />
}

export default AccountInfoForm
