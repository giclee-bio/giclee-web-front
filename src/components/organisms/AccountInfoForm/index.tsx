import AccountInfoForm from './AccountInfoForm'

export default AccountInfoForm
