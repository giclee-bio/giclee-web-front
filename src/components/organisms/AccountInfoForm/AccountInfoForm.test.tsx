import { render } from '@testing-library/react'

import AccountInfoForm from './AccountInfoForm'

describe('AccountInfoForm', () => {
  it('snapshot', async () => {
    const { asFragment } = render(<AccountInfoForm />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
