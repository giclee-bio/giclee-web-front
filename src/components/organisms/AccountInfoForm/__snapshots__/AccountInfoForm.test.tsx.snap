// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountInfoForm snapshot 1`] = `
<DocumentFragment>
  <form
    class="w-64"
  >
    <div
      class="mt-4 first:mt-0"
    >
      <div
        class="flex justify-between text-sm"
      >
        <label
          class="font-bold"
          for="radix-:r0:"
        >
          表示名
        </label>
      </div>
      <input
        class="mt-2 bg-stone-50 h-8 w-full rounded px-3 text-black"
        id="radix-:r0:"
        name="displayName"
        required=""
        title=""
        type="text"
      />
    </div>
    <div
      class="mt-4 first:mt-0"
    >
      <div
        class="flex justify-between text-sm"
      >
        <label
          class="font-bold"
          for="radix-:r1:"
        >
          自己紹介
        </label>
      </div>
      <textarea
        class="mt-2 bg-stone-50 w-full rounded px-3 text-black"
        cols="30"
        id="radix-:r1:"
        name="username"
        required=""
        rows="5"
        title=""
      />
    </div>
    <button
      class="mt-9 block  h-10 rounded text-center font-bold leading-10 w-32 bg-primary text-black"
    >
      更新する
    </button>
  </form>
</DocumentFragment>
`;
